sql: TestScatterRandomizeLeases failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/a0b7cd4ebddf5ebc8f8c2119b119e57688f072f9

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestScatterRandomizeLeases PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=968908&tab=buildLog

```
I181016 05:45:49.282566 95982 storage/replica_command.go:300  [n1,s1,r117/1:/{Table/53/1/9…-Max}] initiating a split of this range at key /Table/53/1/970 [r118]
E181016 05:45:49.290768 97619 storage/queue.go:791  [n1,raftsnapshot,s1,r117/1:/{Table/53/1/9…-Max}] snapshot failed: (n2,s2):2: remote couldn't accept Raft snapshot eb78f510 at applied index 10 with error: [n2,s2],r117: cannot apply snapshot: snapshot intersects existing range [n2,s2,r77/2:/{Table/53/1/5…-Max}]
I181016 05:45:49.294958 95982 storage/replica_command.go:300  [n1,s1,r118/1:/{Table/53/1/9…-Max}] initiating a split of this range at key /Table/53/1/980 [r119]
E181016 05:45:49.301119 97634 storage/queue.go:791  [n1,raftsnapshot,s1,r118/1:/{Table/53/1/9…-Max}] snapshot failed: (n2,s2):2: remote couldn't accept Raft snapshot fcaa08f7 at applied index 10 with error: [n2,s2],r118: cannot apply snapshot: snapshot intersects existing range [n2,s2,r77/2:/{Table/53/1/5…-Max}]
I181016 05:45:49.305099 95982 storage/replica_command.go:300  [n1,s1,r119/1:/{Table/53/1/9…-Max}] initiating a split of this range at key /Table/53/1/990 [r120]
E181016 05:45:49.309042 97552 storage/queue.go:791  [n1,raftsnapshot,s1,r119/1:/{Table/53/1/9…-Max}] snapshot failed: (n2,s2):2: remote couldn't accept Raft snapshot 1b594d18 at applied index 10 with error: [n2,s2],r119: cannot apply snapshot: snapshot intersects existing range [n2,s2,r77/2:/{Table/53/1/5…-Max}]
I181016 05:45:49.353548 97626 util/stop/stopper.go:537  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W181016 05:45:49.353730 96058 storage/raft_transport.go:583  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
I181016 05:45:49.353781 97624 util/stop/stopper.go:537  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181016 05:45:49.353901 97625 util/stop/stopper.go:537  quiescing; tasks left:
2      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W181016 05:45:49.354768 95615 storage/raft_transport.go:583  [n1] while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
W181016 05:45:49.354885 95618 storage/raft_transport.go:583  [n1] while processing outgoing Raft queue to node 3: EOF:
W181016 05:45:49.354939 96110 storage/raft_transport.go:583  [n2] while processing outgoing Raft queue to node 3: EOF:
W181016 05:45:49.355008 95607 storage/raft_transport.go:583  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = context canceled:
W181016 05:45:49.355069 95376 gossip/gossip.go:1496  [n3] no incoming or outgoing connections
W181016 05:45:49.355437 95161 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
W181016 05:45:49.355793 95636 storage/raft_transport.go:583  [n2] while processing outgoing Raft queue to node 1: EOF:
I181016 05:45:49.356583 97626 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181016 05:45:49.356652 97624 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181016 05:45:49.356705 97624 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
I181016 05:45:49.356809 97626 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
I181016 05:45:49.356945 97625 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181016 05:45:49.357080 97625 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
--- FAIL: TestScatterRandomizeLeases (5.27s)
	scatter_test.go:88: expected 100 ranges, got 99

```

--- a/pkg/sql/scatter_test.go
+++ b/pkg/sql/scatter_test.go
@@ -24,6 +24,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
+	"github.com/cockroachdb/cockroach/pkg/storage/storagebase"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
@@ -39,7 +40,17 @@ func TestScatterRandomizeLeases(t *testing.T) {
 	}
 
 	const numHosts = 3
-	tc := serverutils.StartTestCluster(t, numHosts, base.TestClusterArgs{})
+
+	// Prevent the merge queue from immediately discarding our splits. This is
+	// more foolproof than changing the cluster setting because the cluster
+	// setting change has to be propagated to all nodes via gossip so there's
+	// still a small chance that a non-gateway node will try a merge after we
+	// change the setting.
+	var testClusterArgs base.TestClusterArgs
+	testClusterArgs.ServerArgs.Knobs.Store = &storagebase.StoreTestingKnobs{
+		DisableMergeQueue: true,
+	}
+	tc := serverutils.StartTestCluster(t, numHosts, testClusterArgs)
 	defer tc.Stopper().Stop(context.TODO())
 
 	sqlutils.CreateTable(
@@ -51,7 +62,8 @@ func TestScatterRandomizeLeases(t *testing.T) {
 
 	r := sqlutils.MakeSQLRunner(tc.ServerConn(0))
 
-	// Prevent the merge queue from immediately discarding our splits.
+	// Even though we disabled merges via the store testing knob, we must also
+	// disable the setting in order for manual splits to be allowed.
 	r.Exec(t, "SET CLUSTER SETTING kv.range_merge.queue_enabled = false")
 
 	// Introduce 99 splits to get 100 ranges.

